prow/jira/jira.go

```diff
@@ -121,9 +139,40 @@ func (u userAgentSettingTransport) RoundTrip(r *http.Request) (*http.Response, e
 	return u.upstream.RoundTrip(r)
 }
 
+type clientUsedTransport struct {
+	used     bool
+	m        sync.Mutex
+	upstream http.RoundTripper
+}
+
+func (c *clientUsedTransport) RoundTrip(r *http.Request) (*http.Response, error) {
+	c.m.Lock()
+	defer c.m.Unlock()
```

Comment:
By using `defer` here we have just made a single-threaded client :)

Also as a nit, even if the `defer` was not containing the `c.upstream` call, it's about 5-10x slower than explicitly unlocking in cases like this. I think we should:

```go
c.m.Lock()
c.used = true
c.m.Unlock()
c.upstream.RoundTrip(r)
```

Reply:
Yes, you are right. I wanted to change this but then forgot. Thanks.

Reply:
Agree re the single threading point. The performance issues `defer` had were mostly fixed some time back btw: https://rakyll.org/inlined-defers/